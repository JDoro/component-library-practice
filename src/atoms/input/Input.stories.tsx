import type {Meta, StoryObj} from '@storybook/react';

import {Input} from "./Input.component.tsx";
import {useArgs} from '@storybook/preview-api';

const meta = {
  title: 'Atoms/Input',
  component: Input,
  tags: ['autodocs'],
  args: {
    label: 'Enter text here',
  },
  render: (args) => {
    const [{ value, onChange }, updateArgs] = useArgs();
    return (
      <Input
        {...args}
        value={value}
        onChange={(value) => {
          onChange?.(value);
          updateArgs({ value });
        }}
      />
    );
  }
} satisfies Meta<typeof Input>;

export default meta;
type Story = StoryObj<typeof meta>;

export const All: Story = {};

