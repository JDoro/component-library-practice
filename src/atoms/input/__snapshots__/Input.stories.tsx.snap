// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Input All smoke-test 2`] = `
<div class="flex flex-col">
  <label for="input"
         class="text-md mb-1"
  >
    Enter text here
  </label>
  <input id="input"
         type="text"
         class="border-[1px] border-secondary rounded-md"
         value
         style
  >
</div>
`;
